src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+
+  let copy = { ...state };
+  const story = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        copy = {};
+        break;
+
+      case 'addProperties':
+        copy = { ...copy, ...action.extraData };
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete copy[key];
+        });
+        break;
+
+      default:
+        break;
+    }
+    story.push({ ...copy });
```

Comment:
It is better to push a copy of the object to the array at the end of each loop cycle, but outside of the `switch` block, after the current action is processed. This is done correctly here.